vscode-ext/src/StatusBarComponent.ts

```diff
@@ -38,36 +48,156 @@ export class StatusBarComponent {
+
+        verifier.context.subscriptions.push(vscode.workspace.onDidOpenTextDocument(() => {
+            this.render();
+        }));
     }
 
     render() {
         if (this.enabled) {
-            switch (this.status) {
+
+            switch (this.serverStatus) {
+                case ServerStatus.NotStarted:
+                    this.view.text = "$(debug-start) Caesar Inactive";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
+                    break;
                 case ServerStatus.Stopped:
-                    this.view.text = "$(debug-stop) Et tu, Brute?";
+                    this.view.text = "$(debug-start) Et tu, Brute?";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
                     break;
                 case ServerStatus.FailedToStart:
                     this.view.text = "$(warning) Failed to start Caesar";
+                    this.view.command = "caesar.startServer";
+                    this.tooltipMenuText = stoppedTooltipMenu;
                     break;
                 case ServerStatus.Starting:
-                    this.view.text = "$(sync~spin) Starting Caesar...";
+                    this.view.text = "$(loading~spin) Starting Caesar...";
+                    this.view.command = "caesar.showOutput"
                     break;
                 case ServerStatus.Ready:
-                    this.view.text = "$(check) Caesar Ready";
+                    this.handleReadyStatus();
+                    this.view.command = "caesar.showOutput";
+                    this.tooltipMenuText = runningTooltipMenu;
                     break;
                 case ServerStatus.Verifying:
                     this.view.text = "$(sync~spin) Verifying...";
-                    break;
-                case ServerStatus.Finished:
-                    this.view.text = "$(check) Verified";
+                    this.view.command = "caesar.showOutput"
                     break;
             }
+
+            this.renderTooltip();
             this.view.show();
         } else {
             this.view.hide();
         }
     }
+
+    private renderTooltip() {
+        if (this.tooltipStatusText.value === "") {
+            this.view.tooltip = this.tooltipMenuText;
+            this.view.tooltip.isTrusted = true;
+            return;
+        }
+        this.view.tooltip = new vscode.MarkdownString(this.tooltipMenuText.value + "\n\n --- \n" + this.tooltipStatusText.value, true);
+        this.view.tooltip.isTrusted = true;
+    }
+
+
+    private handleReadyStatus() {
+        let tooltipString = new vscode.MarkdownString("", true);
+
+        let someError = false;
+        let someVerified = false;
+
+        const editorsWithResults = vscode.window.visibleTextEditors.filter((editor) => {
+            return editor.document.languageId === "heyvl" // Only HeyVL files
```

Comment:
Why do you need to check for HeyVL files? Shouldn't `this.verifyStatus` already only contain just HeyVL files?